spec/lib/msf/ui/console/command_dispatcher/core_spec.rb

```diff
@@ -397,4 +397,96 @@ def set_tabs_test(option)
       end
     end
   end
+
+  describe "#cmd_sessions" do
+    before(:each) do
+      allow(driver).to receive(:active_session=)
+      
+    end
+
+    context "with no sessions" do
+      it "should show an empty table" do
+        core.cmd_sessions
+        expect(@output).to eq([
+          'Active sessions',
+          '===============',
+          '',
+          'No active sessions.'
+        ])
+      end
+    end
+
+    context "filtering with search results" do
+      before do
+        allow(framework).to receive(:sessions).and_return(sessions)
+        allow(double "Session").to receive(:kill)
+      end
+
+      let(:sessions) do 
+        {
+          1 => double('Session', last_checkin: Time.now, type: 'meterpreter', sid:1, sname: "sesh1", info: "info", session_host: "127.0.0.1", tunnel_to_s: "tunnel"),
+          2 => double('Session', last_checkin: (Time.now - 90), type: 'meterpreter', sid:2, sname: "sesh2", info: "info", session_host: "127.0.0.1", tunnel_to_s: "tunnel"),
+          3 => double('Session', last_checkin: Time.now, type: 'java', sid:3, sname: "sesh3", info: "info", session_host: "127.0.0.1", tunnel_to_s: "tunnel")
```

Comment:
For anything timing related, it's probably best to mock time with TimeCop to ensure that your tests are deterministic on every test run. It also provides a nice API for time travel

https://github.com/rapid7/metasploit-framework/blob/1ee7f03a2b5031f88de37af7b135ec4d83a827f2/spec/lib/msf/ui/console/command_dispatcher/db/klist_spec.rb#L195-L202